lib/trento_web/controllers/v1/settings_controller.ex

```diff
@@ -93,4 +94,50 @@ defmodule TrentoWeb.V1.SettingsController do
       })
     end
   end
+
+  operation :update_activity_log_settings,
+    summary: "Updates the Activity Log settings",
+    tags: ["Platform"],
+    request_body:
+      {"ActivityLogSettings", "application/json", Schema.Platform.ActivityLogSettings},
+    responses: [
+      ok:
+        {"Activity Log settings saved successfully", "application/json",
+         Schema.Platform.ActivityLogSettings},
+      unprocessable_entity: Schema.UnprocessableEntity.response()
+    ]
+
+  def update_activity_log_settings(%{body_params: body_params} = conn, _) do
+    %{retention_period: retention_period, retention_period_unit: retention_period_unit} =
+      body_params
+
+    with {:ok, updated_settings} <-
+           ActivityLog.change_retention_period(retention_period, retention_period_unit) do
+      render(conn, "activity_log_settings.json", %{
+        activity_log_settings: updated_settings
+      })
+    end
+  end
+
+  operation :get_activity_log_settings,
+    summary: "Fetches the Activity Log settings",
+    tags: ["Platform"],
+    responses: [
+      ok:
+        {"Activity Log settings fetched successfully", "application/json",
+         Schema.Platform.ActivityLogSettings},
+      not_found: Schema.NotFound.response()
+    ]
+
+  def get_activity_log_settings(conn, _) do
+    case ActivityLog.get_settings() do
+      {:ok, settings} ->
+        render(conn, "activity_log_settings.json", %{
+          activity_log_settings: settings
+        })
+
+      _ ->
+        {:error, :not_found}
```

Comment:
**suggestion:** ‏As far as I understand, here we are mapping the `:activity_log_settings_not_configured` error to `:not_found`. The reason we don't do it in the fallback controller is because we want differentiate the case of this error from receiving the very same in the `update_activity_log_settings` function, right?

Maybe a comment would be worth to explain this to readers?

**nitpick:** Maybe the context is a better place to do such mapping, but it' just about moving code.‏

Reply:
Sure, let me add a comment. :+1: